src/main/scala/uk/gov/hmrc/SbtArtifactory.scala

```diff
@@ -110,35 +112,33 @@ object SbtArtifactory extends sbt.AutoPlugin {
           .deleteVersion(artifactDescription.value, streams.value.log)
           .awaitResult
     },
-    unpublishFromArtifactoryBintrayDistribution := {
-      streams.value.log.info("SbtArtifactoryPlugin - Unpublishing from Artifactory bintray-distribution...")
-      artifactoryConnector(repoKey.value)
-        .deleteBintrayDistributionVersion(artifactDescription.value, bintrayReleasesFolder.value, streams.value.log)
-        .awaitResult
-    },
-    unpublishFromBintray := {
-      streams.value.log.info("SbtArtifactoryPlugin - Deleting from Bintray...")
-        bintrayConnector(repoKey.value)
-          .deleteReleaseOrPluginVersion(artifactDescription.value, bintrayReleasesFolder.value, streams.value.log)
-          .awaitResult
-    },
-    unpublish := Def
-      .sequential(
-        unpublishFromArtifactory,
-        unpublishFromArtifactoryBintrayDistribution,
-        unpublishFromBintray
-      ).value,
-    distributeToBintray := {
-      streams.value.log.info("SbtArtifactoryPlugin - Distributing to Bintray...")
-        new BintrayDistributor(artifactoryConnector(repoKey.value), streams.value.log)
-        .distributePublicArtifact(artifactDescription.value)
-        .awaitResult
-    },
-    publishAndDistribute := Def
-      .sequential(
-        publish,
-        distributeToBintray
-      ).value
+    unpublishFromBintray := Def.taskDyn({
+      if (artifactDescription.value.publicArtifact) {
+        streams.value.log.info("SbtArtifactoryPlugin - Unpublishing from Bintray...")
+        bintrayUnpublish.toTask
+      } else {
+        Def.task {
+          streams.value.log.info("SbtArtifactoryPlugin - Nothing to unpublish from Bintray...")
+        }
+      }
+    }).result.value,// fail silently
+    unpublish := Def.sequential(
+      unpublishFromArtifactory,
+      unpublishFromBintray
+    ).value,
+    publishToBintray := Def.taskDyn({
+      if(artifactDescription.value.publicArtifact) {
+        streams.value.log.info("SbtArtifactoryPlugin - Distributing to Bintray...")
+        bintrayRelease
+          .dependsOn(publishTask(publishConfiguration, deliver))
+          .dependsOn(bintrayEnsureBintrayPackageExists, bintrayEnsureLicenses)
+      } else {
+        Def.task{
+          streams.value.log.info("SbtArtifactoryPlugin - Not distributing to Bintray...")
+        }
+      }
+    }).value,
```

Comment:
Does the publishToBintray also fail silently?

Reply:
No it won't fail silently. I have intentionally set it so we don't fail silently on publishing to Bintray. We might want to do this after we've sent out comms around using the artifactory resolver but atm I think this is the right approach. It's `result.value` that makes it fail silently by the way.